library_generation/utilities.sh

```diff
@@ -76,16 +77,15 @@ remove_grpc_version() {
 download_gapic_generator_pom_parent() {
   local gapic_generator_version=$1
   if [ ! -f "gapic-generator-java-pom-parent-${gapic_generator_version}.pom" ]; then
-    if [[ "${gapic_generator_version}" == *"-SNAPSHOT" ]]; then
-      # copy a SNAPSHOT version from maven local repository.
-      copy_from "$HOME/.m2/repository/com/google/api/gapic-generator-java-pom-parent/${gapic_generator_version}/gapic-generator-java-pom-parent-${gapic_generator_version}.pom" \
+    # first, try to fetch the generator pom locally
+    bash -c "source ${util_script_dir}/utilities.sh && copy_from \"$HOME/.m2/repository/com/google/api/gapic-generator-java-pom-parent/${gapic_generator_version}/gapic-generator-java-pom-parent-${gapic_generator_version}.pom\" \
```

Comment:
I added this to load the functions into the secondary shell. This is all with the intent to capture the [exit code](https://github.com/googleapis/sdk-platform-java/blob/e2bf12e85777b3c78e1e01829f2018bb8fb22484/library_generation/utilities.sh#L177-L189) in case the local search doesn't work.

Reply:
Since we always try to copy files locally before downloading from maven central, I think we can modify `copy_from` not failing but return a value to indicate whether copy is success or not.

Also, we need to modify the error message in `download_fail` to indicate the file is not found neither locally or in maven central should the download fail.

Reply:
Thanks for pointing this out, makes sense. I modified `copy_from` to return (`echo`) a value instead. I added a message to indicate an external repo attempt will be made in case it wasn't found locally. I also created a more generic function for the pom parent and the jar.